Library/Homebrew/utils/bottles.rb

```diff
@@ -59,10 +59,38 @@ def formula_contents(bottle_file,
           name: resolve_formula_names(bottle_file)[0])
         bottle_version = resolve_version bottle_file
         formula_path = "#{name}/#{bottle_version}/.brew/#{name}.rb"
-        contents = Utils.popen_read "tar", "-xOzf", bottle_file, formula_path
+        command = tar_command "xOf", bottle_file, formula_path
+        receipt_file = `#{command}`
         raise BottleFormulaUnavailableError.new(bottle_file, formula_path) unless $CHILD_STATUS.success?
         contents
       end
+
+      def install_dependencies(compression_type, ignore_tar: false)
+        return unless compression_type == :xz
+        return unless ignore_tar || DependencyCollector.tar_needs_xz_dependency?
+        f = Formula["xz"]
+        return if f.optlinked?
+        fi = FormulaInstaller.new(f)
```

Comment:
I don't think `FormulaInstaller` should be invoked here. Where this happens as part of a `brew install` it should be installed as a dependency (and could use `FormulaInstaller` if necessary). Otherwise it should error out and request the user run `brew install xz` manually themselves. We should be optimising primarily for the macOS support versions case.